tensorboard/webapp/widgets/histogram/histogram_component.ts

```diff
@@ -76,6 +76,34 @@ export interface TooltipData {
   };
 }
 
+const LARGE_NUMBER = 10000;
+const SMALL_NUMBER = 0.001;
+
+const d3LargeFormatter = d3.format('.2~s');
+const d3SmallFormatter = d3.format('.2~e');
+const d3MiddleFormatter = d3.format('.4~r');
+
+function formatNumberShort(x: number | {valueOf(): number} ): string {
+    /**
+    *  Formats very large nubmers using SI notation, very small numbers
+    *  using exponential notation, and mid-sized numbers using their
+    *  natural printout.
+    */
+    if (x === 0) {
+        return '0';
+    }
+    const absNum = Math.abs(x as number);
+    if (absNum >= LARGE_NUMBER){
+        return d3LargeFormatter(x);
+    }
+    if (absNum < SMALL_NUMBER){
+        return d3SmallFormatter(x);
+    }
+    return d3MiddleFormatter(x);
```

Comment:
This is very nitpicky but it's a little strange that this is ordered `LARGE_NUMBER`, `SMALL_NUMBER`, `MEDIUM_NUMBER`.

Perhaps consider gating around a series minimums.
```suggestion
    if (absNum >= LARGE_NUMBER){
        return d3LargeFormatter(x);
    }
    if (absNum >= SMALL_NUMBER){
        return d3MiddleFormatter(x);
    }
    return d3SmallFormatter(x);
```

If taking this suggestion you might also consider renaming `SMALL_NUMBER` to `MEDIUM_NUMBER`.